File: scratch/search_in_buyer.js
const fs = require('fs');

function searchFile(filepath, pattern) {
  console.log(`Searching for '${pattern}' in {filepath}...`);
  const content = fs.readFileSync(filepath, 'utf8');
  const lines = content.split('\n');
  const regex = new RegExp(pattern, 'i');
  lines.forEach((line, idx) => {
    if (regex.test(line)) {
      console.log(`L${idx + 1}: ${line.trim().slice(0, 150)}`);
    }
  });
}

searchFile('js/buyer.js', 'initDetailPage');
searchFile('js/buyer.js', 'detail-content');
searchFile('js/buyer.js', 'quay lại');
searchFile('js/buyer.js', 'back-btn');
searchFile('js/buyer.js', 'href');

File: scratch/search_in_buyer.py
import re

def search_file(filepath, pattern):
    print(f"Searching for '{pattern}' in {filepath}...")
    with open(filepath, 'r', encoding='utf-8', errors='ignore') as f:
        for idx, line in enumerate(f, 1):
            if re.search(pattern, line, re.IGNORECASE):
                print(f"L{idx}: {line.strip()[:150]}")

search_file('js/buyer.js', 'AI_REC_SYSTEM')
search_file('js/buyer.js', 'explain')
search_file('js/buyer.js', 'xai')
